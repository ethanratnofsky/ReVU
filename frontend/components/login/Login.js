import { Image, KeyboardAvoidingView, Text, TextInput, TouchableOpacity, View } from 'react-native';
//import { auth } from './firebase';
import loginStyles from './LoginStyles';
import { useEffect, useState } from 'react';
import { useNavigation } from '@react-navigation/native';

const Login = () => {

    const [email, setEmail] = useState('');
    const [pswd, setPswd] = useState('');
    // const navigation = useNavigation();

    // useEffect(() => {
    //     auth.onAuthStateChanged(user => {
    //         if (user) {
    //             navigate
    //         }
    //     })
    // }, [])


    const handleSignUp = () => {
        // if (!(/@vanderbilt.edu\s*$/).test(email)) {
        //     alert("Email is not a valid Vanderbilt University email.");
        // } else {
        //     auth.createUserWithEmailAndPassword(email, pswd).then(userCredentials => {
        //         const user = userCredentials.user;
        //         console.log("Registering");
        //         console.log(user.uid);
        //         navigation.navigate("Terms and Conditions", {id: user.uid});
        //     }).catch(error => alert(error.message));
        // }
    }

    const handleLogin = () => {
        // auth.signInWithEmailAndPassword(email, pswd).then(userCredentials => {
        //     const user = userCredentials.user;
        //     console.log("Logging in");
        //     console.log(user.uid);
        //     navigation.navigate('Home', {id: user.uid});
        // }).catch(error => alert(error.message));
    }

    return (
        <KeyboardAvoidingView style={loginStyles.container} behavior="padding">
            <Image style={loginStyles.logo} source={require('../../assets/images/revu-logo.png')} />
            <View style={loginStyles.inputContainer}>
                <TextInput
                    placeholder='Email'
                    // value={ }
                    onChangeText={text => setEmail(text)}
                    style={loginStyles.input}
                />
                <TextInput
                    placeholder='Password'
                    // value={ }
                    onChangeText={text => setPswd(text)}
                    style={loginStyles.input}
                    secureTextEntry
                />
            </View>

            <View style={loginStyles.buttonContainer}>
                <TouchableOpacity
                    onPress={handleLogin}
                    style={loginStyles.button}
                >
                    <Text style={loginStyles.buttonText}>Login</Text>
                </TouchableOpacity>
            </View>
            <View style={loginStyles.buttonContainer}>
                <TouchableOpacity
                    onPress={handleSignUp}
                    style={[loginStyles.button, loginStyles.buttonOutline]}
                >
                    <Text style={loginStyles.buttonOutlineText}>Register</Text>
                </TouchableOpacity>
            </View>
        </KeyboardAvoidingView>
    )
}

export default Login;