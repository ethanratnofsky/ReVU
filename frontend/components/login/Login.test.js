import React from 'react';
const renderer = require('react-test-renderer');
import Login from './Login';

describe('<Login />', () => {
  it('has 4 children', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree.children.length).toBe(4);
  });
});

it('renders correctly', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });