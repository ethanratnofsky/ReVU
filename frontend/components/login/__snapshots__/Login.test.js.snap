// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#1C1C1C",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 119,
        "marginBottom": 50,
        "width": 250,
      }
    }
  />
  <View
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "marginBottom": 4,
          "marginTop": 5,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "marginBottom": 4,
          "marginTop": 5,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 7,
        "width": "60%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#CFAE70",
          "borderRadius": 10,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "700",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 7,
        "width": "60%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#CFAE70",
          "borderRadius": 10,
          "borderWidth": 2,
          "justifyContent": "center",
          "marginTop": 5,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1C1C1C",
            "fontSize": 16,
            "fontWeight": "700",
          }
        }
      >
        Register
      </Text>
    </View>
  </View>
</View>
`;
